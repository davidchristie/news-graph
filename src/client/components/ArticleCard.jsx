// import moment from 'moment'
import React, { PropTypes } from 'react'
import { Link } from 'react-router'

// function footer (publishedAt) {
//   const date = moment(publishedAt).format('MMMM Do YYYY, h:mm:ss a')
//   if (publishedAt) {
//     return (
//       <div className='card-footer'>
//         {date}
//       </div>
//     )
//   }
// }

export default class ArticleCard extends React.Component {
  render () {
    const article = this.props.article
    const url = article.url
    const title = article.title
    const description = article.description
    const image = article.thumbnail_url ? <img className='card-img-top img-fluid' src={article.thumbnail_url} alt='Card image cap' /> : null
    return (
      <div className='card'>
        {image}
        <div className='card-block'>
          <Link to={`/articles?url=${url}`}>
            <h4 className='card-title'>{title}</h4>
          </Link>
          <p className='card-text'>{description}</p>
        </div>
      </div>
    )
  }
}

ArticleCard.propTypes = {
  article: PropTypes.shape({
    // id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired
}
