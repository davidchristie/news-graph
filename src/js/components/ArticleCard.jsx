import moment from 'moment'
import React, { PropTypes } from 'react'
import { Link } from 'react-router'

function footer (publishedAt) {
  const date = moment(publishedAt).format('MMMM Do YYYY, h:mm:ss a')
  if (publishedAt) {
    return (
      <div className="card-footer">
        {date}
      </div>
    )
  }
}

export default class ArticleCard extends React.Component {
  render () {
    const article = this.props.article
    const id = article.id
    const title = article.title
    const description = article.description
    // const url = article.url
    const urlToImage = article.urlToImage
    const publishedAt = article.publishedAt
    return (
      <div className="card">
        <img className="card-img-top img-fluid" src={urlToImage} alt="Card image cap"/>
        <div className="card-block">
          <Link to={`article/${id}`}>
            <h4 className="card-title">{title}</h4>
          </Link>
          <p className="card-text">{description}</p>
        </div>
        {footer(publishedAt)}
      </div>
    )
  }
}

ArticleCard.propTypes = {
  article: PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired
}
