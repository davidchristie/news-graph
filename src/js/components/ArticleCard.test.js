/* eslint-disable no-undef */

import React from 'react'
import renderer from 'react-test-renderer'

import ArticleCard from './ArticleCard'

const exampleArticle = {
  id: 1,
  url: 'http://www.nature.com/news/ancient-retroviruses-emerged-half-a-billion-years-ago-1.21274'
}

test('ArticleCard renders correctly', () => {
  const tree = renderer.create(
    <ArticleCard article={exampleArticle} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
