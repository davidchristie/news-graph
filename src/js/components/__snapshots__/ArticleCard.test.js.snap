exports[`test ArticleCard renders correctly 1`] = `
<div
  className="card">
  <img
    alt="Card image cap"
    className="card-img-top img-fluid"
    src={undefined} />
  <div
    className="card-block">
    <a
      onClick={[Function]}
      style={Object {}}>
      <h4
        className="card-title" />
    </a>
    <p
      className="card-text" />
  </div>
</div>
`;
